components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/ServiceURLBuilder.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.core;
+
+import org.wso2.carbon.identity.core.model.ServiceURL;
+
+/**
+ * URL Builder service interface.
+ */
+public interface ServiceURLBuilder {
+
+    static ServiceURLBuilder create() {
+
+        return new DefaultServiceURLBuilder();
+    }
+
+    /**
+     * This method is called to add the URL context to the builder.
+     *
+     * @param context URL context.
+     * @return {@link ServiceURLBuilder}.
+     */
+    ServiceURLBuilder addContext(String context);
```

Comment:
I think multiple context paths can be added and you should be building the requestURI

Reply:
Didn't get what's meant by 'multiple context paths' here. Is it the web context and the proxy context?

Reply:
multiple paths.
like addContext(oauth2).addContext(token) should add /oauth2/token

Reply:
Maybe it's better to say 'addPath(String... paths)' and provide the capability to add multiple paths with a comma separated string

Reply:
I think addPath approach is better